### ISSUE TITLE
API error handling refactor breaks validation and has no stacktraces

### ISSUE THREAD

**Issue by datitisev** [flarum/core#1865]
Introduced with #1843.

1. JSON API errors no longer contain the stack trace
2. Illuminate Validation exception (maybe the Flarum one as well) is completely broken.

The response is always
```json
{
  "errors": [
    {
      "status": "422",
      "code": "validation_error"
    }
  ]
}
```

I tried adding the exception details (which for some reason aren't even passed to the response right now), but that made the response look like this:
```json
{
  "errors": [
    {
      "status": "422",
      "code": "validation_error",
      "detail": [
        {
          "detail": "The content field is required.",
          "source": {
            "pointer": "/data/attributes/content"
          }
        }
      ]
    }
  ]
}
```

When it should have the `detail` and `source` directly on the error object... at least that's how the JS views it anyway. We could modify how the JS validation handler finds the details, but they still need to be added to the response.

**Comment by franzliedke** [comment]
Fixed this, including a regression test.

@datitisev Can you expand on this part?

> JSON API errors no longer contain the stack trace

I suppose you're referring to [this piece of code from the json-api library](https://github.com/tobyzerner/json-api-php/blob/355b213927a1412ede81b67ac2a5605753382379/src/Exception/Handler/FallbackExceptionHandler.php#L60-L62)?

**Comment by datitisev** [comment]
@franzliedke Errors thrown with debug mode would contain the stack trace for the exception in the JSON response. They no longer do, which means (1) re-adding the debug button in the JS at some point would be pointless, and (2) requires looking at log files which is more cumbersome. 

After looking through issues reported earlier, yeah, looks the JSON API isn't adding the `detail` with the exception... it should for even 404 errors iirc.

---

TL;DR: yes, seems like it's not being passed or something ?

**Comment by franzliedke** [comment]
Stacktraces are now restored as well. :v:

**Comment by datitisev** [comment]
Yay 🎉. Tests failed because you added the `detail` property back though :P

**Comment by franzliedke** [comment]
Fixed. Dang, I should have run all the tests and not just the ones from yesterday. :laughing:
